src/main/java/org/gridsuite/securityanalysis/server/repositories/CommonLimitViolationRepository.java

```diff
@@ -46,9 +48,23 @@ default Specification<T> getParentsSpecifications(
                 // user filters on OneToMany collection - needed here to filter main entities that would have empty collection when filters are applied
                 childrenFilters
                     .forEach(filter -> addPredicate(criteriaBuilder, root.get("contingencyLimitViolations"), predicates, filter));
-            } else {
+            } else if (isSubjectLimitViolations) {
                 // filter parents with empty children even if there isn't any filter
                 predicates.add(criteriaBuilder.isNotEmpty(root.get("contingencyLimitViolations")));
+            } else {
+                Predicate isNotEmptyPredicate = criteriaBuilder.isNotEmpty(root.get("contingencyLimitViolations"));
+                // Add condition for status "CONVERGED" and contingencyLimitViolations isNotEmpty
+                Predicate convergedCondition = criteriaBuilder.and(criteriaBuilder.equal(root.get("status"),
+                        LoadFlowResult.ComponentResult.Status.CONVERGED.toString()), isNotEmptyPredicate);
+                // Add condition for status "MAX_ITERATION_REACHED" or "FAILED" or "SOLVER_FAILED"
+                Predicate maxIterationOrFailedCondition = criteriaBuilder.or(criteriaBuilder.equal(root.get("status"),
+                        LoadFlowResult.ComponentResult.Status.MAX_ITERATION_REACHED.toString()),
+                        criteriaBuilder.equal(root.get("status"), LoadFlowResult.ComponentResult.Status.FAILED.toString()),
+                        criteriaBuilder.equal(root.get("status"), LoadFlowResult.ComponentResult.Status.SOLVER_FAILED.toString()));
+
+                // Combine the conditions
+                Predicate finalCondition = criteriaBuilder.or(convergedCondition, maxIterationOrFailedCondition);
```

Comment:
Here we have this condition :
(children is not empty AND status === CONVERGED) OR 
(status === MAX_ITERATION_REACHED OR 
status ===  FAILED OR
status === SOLVER_FAILED)

I think it would be easier to read and maintain to use this condition : 
children is not empty OR status != CONVERGED

Reply:
done